include/vcpkg/base/cmd-parser.h

```diff
@@ -23,4 +31,144 @@ namespace vcpkg
 
         std::string m_str;
     };
-}
\ No newline at end of file
+
+    std::vector<std::string> convert_argc_argv_to_arguments(int argc, const CommandLineCharType* const* const argv);
+    ExpectedL<Unit> replace_response_file_parameters(std::vector<std::string>& inputs,
+                                                     const ILineReader& response_file_source);
+
+    struct SwitchName
+    {
+        std::string switch_name;
+        StabilityTag stability;
+        bool operator<(const SwitchName& rhs) const;
+    };
+
+    struct CmdParser
+    {
+        CmdParser();
+        explicit CmdParser(View<std::string> inputs);
+        explicit CmdParser(std::vector<std::string>&& inputs);
+        CmdParser(const CmdParser&);
+        CmdParser(CmdParser&&);
+        CmdParser& operator=(const CmdParser&);
+        CmdParser& operator=(CmdParser&&);
+
+        // Parses a switch from the input named switch_name, and stores the value if
+        // encountered into value. Returns true if the switch was encountered.
+        // Emits an error if the switch is encountered more than once.
+        bool parse_switch(StringView switch_name, StabilityTag stability, bool& value);
```

Comment:
I tried this but I'm not a fan:

<img width="751" alt="example switchname diff" src="https://user-images.githubusercontent.com/1544943/225180651-22a69050-cd01-4da2-b18e-e82d4d9be795.png">

`SwitchName` is here mainly so it can be a key in the options table map, not because I expect callers of this stuff to interact with it.

https://github.com/BillyONeal/vcpkg-tool/commit/84a9aff758d17e212b201cc3a41720dbc5431595

Reply:
After in person discussion we've agreed to defer this until we can StringLiteralize.